src/libYARP_OS/src/PortReaderBufferBase.cpp

```diff
@@ -572,35 +572,24 @@ bool PortReaderBufferBase::acceptObjectBase(PortReader *obj,
             mPriv->consumeSema.wait();
         }
     }
-    bool ok = true;
-    if (ok) {
-        reader->setExternal(obj, wrapper);
 
-        mPriv->stateMutex.lock();
-        bool pruned = false;
-        if (mPriv->ct>0 && mPriv->prune) {
-            PortReaderPacket *readerPacket =
-                mPriv->dropContent();
-            pruned = (readerPacket!=nullptr);
-        }
-        //mPriv->configure(reader, false, true);
-        mPriv->pool.addActivePacket(reader);
-        mPriv->ct++;
-        mPriv->stateMutex.unlock();
-        if (!pruned) {
-            mPriv->contentSema.post();
-        }
-        //YARP_ERROR(Logger::get(), ">>>>>>>>>>>>>>>>> adding data");
-    } else {
-        mPriv->stateMutex.lock();
-        mPriv->pool.addInactivePacket(reader);
-        mPriv->stateMutex.unlock();
-        //YARP_ERROR(Logger::get(), ">>>>>>>>>>>>>>>>> skipping data");
+    reader->setExternal(obj, wrapper);
 
-        // important to give reader a shot anyway, allowing proper closing
-        YARP_DEBUG(Logger::get(), "giving PortReaderBuffer chance to close");
+    mPriv->stateMutex.lock();
+    bool pruned = false;
+    if (mPriv->ct>0 && mPriv->prune) {
+        PortReaderPacket *readerPacket =
+            mPriv->dropContent();
+        pruned = (readerPacket!=nullptr);
+    }
+    //mPriv->configure(reader, false, true);
+    mPriv->pool.addActivePacket(reader);
+    mPriv->ct++;
+    mPriv->stateMutex.unlock();
+    if (!pruned) {
```

Comment:
I'm I am afraid that
```c++
bool ok = true;
if (ok) { ...
}
else { ...
}
```
one time had sense, was it inside a loop and `ok` were updated?
Obviously your fix is right, I'm just worried about the possible hidden bug.

Reply:
We will never know why that piece of code is like that. The comment on the function 
```
/// Careful!  merge with ::read -- very similar code
/// Until merge, don't change one without looking at other :-(
```
made me think about a copy/paste and change where needed. In the `::read()` function that `ok` variable is set with a result of an operation, maybe here it was not the case but to keep the same structure of the code it was directly set to true.

Reply:
You are right it has been completely copy-pasted, so I think we can safely remove this useless `if-else`

Reply:
Still not 100% convinced that it is a good idea to merge this change, but anyway...